src/Model/ModelManager.php

```diff
@@ -346,10 +346,19 @@ public function batchDelete(string $class, BaseProxyQueryInterface $query): void
         }
 
         try {
+            if ([] !== $query->getQueryBuilder()->getDQLPart('join')) {
+                $rootAlias = current($query->getQueryBuilder()->getRootAliases());
+
+                // Distinct is needed to iterate, even if group by is used
+                // @see https://github.com/doctrine/orm/issues/5868
+                $query->getQueryBuilder()->distinct();
+                $query->getQueryBuilder()->select($rootAlias);
+            }
```

Comment:
Does it need to be in the try catch ?

The less code we have inside, the better we understand what we want to catch (?)

Reply:
I moved code outside the try catch.